packages/ember-glimmer/lib/syntax/outlet.js

```diff
@@ -201,7 +201,7 @@ const TOP_LEVEL_MANAGER = new TopLevelOutletComponentManager();
 
 class OutletComponentManager extends AbstractOutletComponentManager {
   create(definition, args, dynamicScope) {
-    let outletStateReference = dynamicScope.outletState = dynamicScope.outletState.get(definition.outletName);
+    let outletStateReference = dynamicScope.outletState = dynamicScope.outletState.get('outlets').get(definition.outletName);
```

Comment:
I guess we don't _have_ to use an `EmberPathReference` if we think the `ChildOutletStateReference` is simpler and faster (and looking at your test reminds me that it could have been _any_ PathReference, in your test it would be a HelperReference). I think the key is that we cannot rely on anything that is not strictly part of the usual `PathReference` interface – so things like `isTopLevel` raises a red-flag.